Fix #12118: Tooltip: escape selector for mouseTarget.closest

redddcyclone opened the pull request:
Fix #12118

This PR fixes an issue that happens when the tooltip's ID has characters that should be escaped, like the colon, something that happens frequently on Faces.

redddcyclone committed:
Fix #12118: Tooltip: escape selector for mouseTarget.closest

Fix an issue when the tooltip's ID has characters that should be escaped, like the colon, something that happens frequently on Faces.

melloware commented on primefaces/src/main/resources/META-INF/resources/primefaces/tooltip/tooltip.js:
```diff
@@ -302,7 +302,7 @@ PrimeFaces.widget.Tooltip = PrimeFaces.widget.BaseWidget.extend({
             var previousElement = this.target;
             this.allowHide = !(mouseTarget.is(previousElement) ||
-                               previousElement.attr('aria-describedby') === mouseTarget.closest('#' + this.id).attr('id'));
+                               previousElement.attr('aria-describedby') === mouseTarget.closest('#' + $.escapeSelector(this.id)).attr('id'));
```
```suggestion
                               previousElement.attr('aria-describedby') === mouseTarget.closest(PrimeFaces.escapeClientId((this.id)).attr('id'));
```

redddcyclone replied:
I don't know how ```PrimeFaces.escapeClientId``` works, but wouldn't it remove the required # character?

EDIT: Never mind, I tried it and it works correctly.

melloware replied:
No it adds the `#` and escapes the `:`

```js
/**
         * Creates an ID to a CSS ID selector that matches elements with that ID. For example:
         * ```
         * PrimeFaces.escapeClientId("form:input"); // => "#form\:input"
         * PrimeFaces.escapeClientId("form#input"); // => "#form#input"
         * ```
         *
         * __Please note that this method does not escape all characters that need to be escaped and will not work with arbitrary IDs__
         * @param {string} id ID to convert.
         * @return {string} A CSS ID selector for the given ID.
         */
        escapeClientId : function(id) {
            return "#" + id.replace(/:/g,"\\:");
        },
```

redddcyclone committed:
Fix #12118: Replace escapeSelector and restore old functionality

This replaces $.escapeSelector with PrimeFaces.escapeClientId and restores the functionality of the tooltip not hiding when the mouse moves back to the tooltip's linked component.
